core-js/src/main/javascript/cdf-legacy/queries/coreQueries.js

```diff
@@ -681,7 +681,15 @@
     interfaces: {
       lastResultSet: {
         reader: function(json) {
-          json = eval("(" + json + ")");
+          var str = json;
+          try {
+            json = JSON.parse(str);
+          } catch (err) {
+            if (!(err instanceof SyntaxError)) {
+              throw err;
+            }
+            json = JSON.parse(str.replace(/'/g, '"'));
```

Comment:
@akhayrutdinov @diogofscmariano In this case, because we might have a single quote in the content of a valid string, I think this will create a regression. As an example, change **LegacyQuery-spec.js** line **17**-**20** to:
`var unprocessedDataString = "{\"metadata\":[\"Sales\"],\"values\":[[\"Euro+ Shopping's Channel\",\"914.11\"],[\"Mini Gifts Ltd.\",\"6558.02\"]]}",`
`unprocessedData = {metadata: [{colIndex : 0, colType : "String", colName : "Name"},`
` {colIndex : 1, colType : "String", colName : "Sales"}],`
`resultset: [["Euro+ Shopping's Channel", "914.11"], ["Mini Gifts Ltd.", "6558.02"]]},`
Notice how the single quote in `"Euro+ Shopping's Channel"` is wrongly changed.